### PR TITLE
feat: v-once for component / v-once with v-for

### PR DESCRIPTION
**Pull request by Jevon617**

### CODE REVIEW

File: packages/runtime-vapor/src/component.ts [hunk 1]
```diff
@@ -263,6 +263,7 @@ export function createComponentInstance(
   dynamicSlots: DynamicSlots | null = null,
   // application root node only
   appContext: AppContext | null = null,
+  once: boolean = false,
```
Comment (sxzz) [inline packages/runtime-vapor/src/component.ts:266]: Move `appContext` to the very bottom.

File: packages/compiler-vapor/src/generators/for.ts [hunk 2]
```diff
@@ -67,11 +67,18 @@ export function genFor(
...
     NEWLINE,
     `const n${oper.id} = `,
-    ...genCall(vaporHelper('createFor'), sourceExpr, blockFn, getKeyFn),
+    ...genCall(
+      vaporHelper('createFor'),
+      sourceExpr,
+      blockFn,
+      !!getKeyFn ? getKeyFn : 'void 0',
+      'void 0', // todo: getMemo
+      'void 0', // todo: hydrationNode
```
Comment (sxzz) [inline packages/compiler-vapor/src/generators/for.ts:79]: If `once` is false, then we can omit these

File: packages/compiler-vapor/src/generators/component.ts [hunk 3]
```diff
@@ -40,8 +40,11 @@ export function genCreateComponent(
     ...genCall(
       vaporHelper('createComponent'),
       tag,
-      rawProps || (isRoot ? 'null' : false),
-      isRoot && 'true',
+      rawProps || 'null',
+      'null', // slots
+      'null', // dynamicSlots
+      isRoot ? 'true' : 'false',
```
Comment (sxzz) [inline packages/compiler-vapor/src/generators/component.ts:46]: If `once` is false, then we can omit these

**Comment by Jevon617** [comment]
[Demo](https://deploy-preview-201--vapor-repl.netlify.app/#__VAPOR__eNp9VE2P0zAQ/Ssjc0grsYlgOUXZlQDtAQ6wAsQplyiZtl4c27KdbFGU/874I2kWQStV8nz6vRm/TOy91vk4ICtZZVvDtQOLbtAgGnm8q5mzNbuvJe+1Mg4mMHiAGQ5G9ZBRWbaGWtXr5M+Lj2T4rhRulbQOenuEO1+8y968vc32tayKeB01J8Nhr0XjkCyAquNjOAC0J4KBvryEiks9OHC/NXpgeHY1g/GmVx0KclBOgEr1RWzgz+PNQRl45u5ERyVbLJcbkh0zqJ4TBuDS30V9pgmCY55fdgslSga6JYRegXmKlFS9YIGCqqpiQ429Zlx2eM5PrhebiSdKxGQQmEjE3zr31iD1+NmQSRvbrqAYvZMm/VeNT1v2kZa8SXrZb0f/fd6rQbpd9qrROtuv+wmz4h3hIz+Bi/MgKs7Sbg/8mD9ZJYnN5LvXzI+DCzRfteO0+5qVECI+1gihnj8HnzMDvl787QnbX//wP9mz99Xs0aBFM9Jw1phrzBHpCfjww/cv4TmswWWUV4Lf0CoxeIwx7cMgO4K9yQtoP4Vpcnn8YR/ODqVdSHmgPnMO+TWj+fpn/z/qF7i3+btQV8uZprhI5boAzTUJUjCqTBulLemswwOX+Oitagrasc4Qhfl+t79kC9U2YlFlMEiXFL0ok4zguDxhspI+vaJJOD3oxqCM+ie6UwSR+/CqnVCVaq+LOMBIhNNFEaY9qUF06XsQ7on+7R1brZHJ5j/pcaKC)

**Review by sxzz — APPROVED** [review]

**Pull request merged**